### PR TITLE
ES-2012: Create secrets pre-upgrade as well as pre-install

### PR DESCRIPTION
**Pull request by anton-subbotin**
### [ES-2012](https://r3-cev.atlassian.net/browse/ES-2012)

Tested on the `5.2.0` stream (`5.1` to `5.2` upgrade path) with `upgradeTestsPipeline`, modified deliberately to reproduce the issue:
- The pipeline change sets username via `"value:"` for the workers' Config and State Manager DBs: https://github.com/corda/corda-shared-build-pipeline-steps/commit/9261f282e62b65604932afb5d554d841a1b86a1e
- With the above only, upgrading to `5.2.0`, the original issue is reproducible. Failed helm upgrade: [https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda5-upgrade-tests/detail/asubbotin%2FES-2012%2Fupgrade-secret-creation-5.2/4/pipeline](https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda5-upgrade-tests/detail/asubbotin%2FES-2012%2Fupgrade-secret-creation-5.2/4/pipeline)
```
55s         Warning   FailedMount              pod/corda-bob-crypto-worker-5fc86c9787-xw8dh                   MountVolume.SetUp failed for volume "key-rotation-volume" : secret "corda-bob-crypto-key-rotation-db" not found
55s         Warning   FailedMount              pod/corda-bob-flow-worker-f4b8fd779-2rxj8                      MountVolume.SetUp failed for volume "flow-checkpoint-volume" : secret "corda-bob-flow-flow-checkpoint-db" not found
54s         Warning   FailedMount              pod/corda-bob-rest-worker-5d8b5bdf89-nb2sq                     MountVolume.SetUp failed for volume "key-rotation-volume" : secret "corda-bob-rest-key-rotation-db" not found
... skipped
```
- With the change applied, a `5.2.0-alpha` build was created: https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os/job/asubbotin%2FES-2012%2Fupgrade-secret-creation-5.2/5/
- Successful upgrade - issue is fixed: https://ci02.dev.r3.com/job/Corda5/job/corda5-upgrade-tests/job/asubbotin%2FES-2012%2Fupgrade-secret-creation-5.2/10/

[ES-2012]: https://r3-cev.atlassian.net/browse/ES-2012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: charts/corda/templates/secrets.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@
               ( printf "bootstrap.db.%s" $db )
               ( printf "%s-%s-db" ( include "corda.fullname" $ ) ( kebabcase $db ) )
               ( dict "username" ( dict "required" true ) "password" ( dict "generate" 12 ) )
-              ( dict "cleanup" true )
+              ( dict "cleanup" true "hook" "pre-install" )
```
Comment (anton-subbotin) [inline charts/corda/templates/secrets.yaml:31]: This is not strictly necessary: even if these three secrets are created `pre-upgrade`, they don't affect the upgrade process and are deleted following cleanup policy on hook success. However, when created on upgrade, these secrets have irrelevant generated passwords and effectively are an unnecessary noise.

**Review by wzur-r3 — DISMISSED** [review]
LGTM

File: charts/corda/templates/secrets.yaml (hunk 1 above)
Comment (davidcurrie) [reply charts/corda/templates/secrets.yaml:31]: Does this not apply anywhere that we specify `cleanup` to be `true`? I.e. do we need another parameter?

Comment (anton-subbotin) [reply charts/corda/templates/secrets.yaml:31]: I initially did it the way you suggest, but then thought that it isn't obvious enough for the code reader. Speaking generally, hook selection does not necessarily need to depend on the cleanup policy.

Comment (davidcurrie) [reply charts/corda/templates/secrets.yaml:31]: That is true. Setting aside the question of whether there should be a separate parameter, there are other places that we specify `true` for `cleanup`. Should they also specify only the `pre-install` hook?

Comment (anton-subbotin) [reply charts/corda/templates/secrets.yaml:31]: You are right; the secrets in question hold the DB credentials used during bootstrapping only, and creating them during upgrade process doesn't make any sense.

Comment (anton-subbotin) [reply charts/corda/templates/secrets.yaml:31]: Done

**Review by davidcurrie — APPROVED** [review]
LGTM 👍

**Pull request merged**